libs/agent-sm/agent/src/main/java/org/opensearch/javaagent/RuntimeHaltInterceptor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.javaagent;
+
+import org.opensearch.javaagent.bootstrap.AgentPolicy;
+
+import java.lang.StackWalker.Option;
+import java.security.Policy;
+import java.util.List;
+
+import net.bytebuddy.asm.Advice;
+
+/**
+ * {@link Runtime#halt} interceptor
+ */
+public class RuntimeHaltInterceptor {
+    /**
+     * SystemExitInterceptor
+     */
+    public RuntimeHaltInterceptor() {}
+
+    /**
+     * Interceptor
+     * @param code exit code
+     * @throws Exception exceptions
+     */
+    @Advice.OnMethodEnter
+    @SuppressWarnings("removal")
+    public static void intercept(int code) throws Exception {
+        final Policy policy = AgentPolicy.getPolicy();
+        if (policy == null) {
+            return; /* noop */
+        }
+
+        final StackWalker walker = StackWalker.getInstance(Option.RETAIN_CLASS_REFERENCE);
+        final Class<?> caller = walker.getCallerClass();
+        final List<Class<?>> chain = walker.walk(new StackCallerClassChainExtractor());
+
+        if (AgentPolicy.isChainThatCanExit(caller, chain) == true) {
```

Comment:
Why not flip this around?

```
if (!AgentPolicy.isChainThatCanExit(caller, chain)) {
    throw new SecurityException("The class " + caller + " is not allowed to call Runtime::halt(" + code + ")");
}
```

There's no need for an explicit return afterwards if the condition is flipped.

Reply:
I see this was changed down below in SystemExitInterceptor. Would it be fair to say this is deny-by-default and only pass if explicitly granted?

Reply:
Yes, as per https://github.com/opensearch-project/OpenSearch/pull/17757#discussion_r2023885120, thanks!